torch/autograd/profiler.py

```diff
@@ -505,12 +547,27 @@ def _privateuse1_memory_usage(mem_record):
                             f_evt.device_index,
                             f_evt.time_range.end - f_evt.time_range.start,
                         )
+                    elif f_evt.device_type == DeviceType.XPU:
+                        fe.append_xpu_kernel(
+                            f_evt.name,
+                            f_evt.device_index,
+                            f_evt.time_range.end - f_evt.time_range.start,
+                        )
+                    elif f_evt.device_type == DeviceType.PrivateUse1:
+                        fe.append_privateuse1_kernel(
+                            f_evt.name,
+                            f_evt.device_index,
+                            f_evt.time_range.end - f_evt.time_range.start,
+                        )
                     elif f_evt.device_type == DeviceType.CPU:
-                        # make sure that 'thread' of a CPU Kineto (e.g. CUDA Runtime) event is associated
+                        # make sure that 'thread' of a CPU Kineto (e.g. CUDA/XPU Runtime) event is associated
                         # with the 'thread' of the corresponding linked PyTorch event to properly track
                         # parents and children
                         f_evt.thread = fe.thread
 
+        # Merge function_events list
+        function_events = function_events.expend(function_events_no_link)
+
```

Comment:
@briancoutinho  Finally review here please. After this line, all things of `function_events` can be used as normal without changes.

Reply:
A mistype here. Have corrected `expend` to `extend`.